### DISCUSSION TITLE
How to mock an imported function?

### DISCUSSION THREAD

**Discussion by dergunovs** [vitest-dev/vitest#1495] [Q&A]
Hi. Very simple question. How to mock a function from ts file in Vue3 component with script setup syntacsis?

Component:
```
<script setup lang="ts">
import { toast } from "@/components/ui/toast"

toast("hi")
</script>
```

In jest i did it that way:
```
let wrapper = shallowMount(Component, {
    global: {
      mocks: { toast: toast() },
    },
  });

const toast = jest.fn();

expect(toast).toBeCalledTimes(1);
```

How to spy on such functions in Vitest? I've tried many variants, but the imported function is not even mocked.
```
1) vi.mock('@/components/ui/toast')
2) vi.mock('toast')
3) vi.mock('@/components/ui/toast', () => {
  return {
    toast: vi.fn(),
  };
});
```

**Comment by sheremet-va** [comment]
This is weird, we had a similar issue in https://github.com/vitest-dev/vitest/issues/1484

It seems something broke. because it worked before 🤔 

Your `vi.mock` variants should have worked

**Reply by dergunovs** [reply]
I can share my vite.config test section. The only unusual thing here is vuetify 3 beta. There are no more external deps in project.

```
test: {
    setupFiles: 'vuetify.config.js',
    deps: { inline: ['vuetify'] },
    globals: true,
    environment: 'jsdom',
    include: ['**/*.spec.js'],
    coverage: { reporter: ['text'], include: ['**/*.vue'] },
  },
```
